### PR TITLE
Constrained operations which rely on ordering to only overload when their values can be totally_ordered

### PR DESCRIPTION
**Pull request by danieltowner**
Not sure whether this is really the right wording, but it captures where suitable wording needs to be added. 

Alternative wording could be to say that `min` is constrained by having `operator<` defined for T, but `simd::min` doesn't actually have to be implemented in terms of `operator<` at all. T::value_type doesn't have to define `operator<` either, and wouldn't be implemented in terms of `operator<(value_type, value_type)` either. The constraint is really on whether it makes sense to compare to value_types in a mathemetical sense, rather than something that actually provides all the right operators, but I'm not sure of the best way to say that.

### CODE REVIEW

File: P1928/wording.tex [hunk 1]
```diff
@@ -1345,6 +1345,8 @@
 \end{itemdecl}
 
 \begin{itemdescr}
+  \pnum\constraints
+  \tcode{totally_ordered<T>} is \tcode{true}
```
Comment (mattkretz) [inline P1928/wording.tex:1349]: `\tcode{T} satisfies \tcode{totally_ordered}.`

And then please add an empty line before the next `\pnum`

I'm not sure whether we also want
```tex
\pnum\expects
\tcode{T} models \tcode{totally_ordered}.
```
NaNs don't *model* the concept. So I believe this precondition would say that passing NaNs triggers UB.

**Review by mattkretz — CHANGES_REQUESTED** [review]
Thanks, good "forward looking" change :wink:.

This change should get a mention at least in `changelog.tex` if not also in the "changes after the TS" section.

File: P1928/wording.tex (hunk 1 above)
Comment (danieltowner) [reply P1928/wording.tex]: Fixed

File: P1928/main.tex [hunk 2]
```diff
@@ -283,6 +283,15 @@ \subsection{Rename \code{simd_mask} reductions}
 
 For a discussion of this topic see \wglink{P1928R3} Section 5.2.
 
+\subsection{Added constraints on operators and functions to match their underlying element types}
+
+Previously some operators (e.g., \code{operator<}) and functions which relied on
+some property of the element type (e.g., \tcode{min} relies on ordering)
+were unconstrained. Operations which were not permitted on individual elements
+were still available in the overload set for simd objects of those types.
+Constraints have been added were necessary to remove such operators and
```
Comment (mattkretz) [inline P1928/main.tex:292]: typo: were -> where

**Review by mattkretz — CHANGES_REQUESTED** [review]
:+1: only a typo left :)

**Pull request merged**